### PR TITLE
Comprehensive PR Template

### PR DESCRIPTION
**Pull request by jeremyk-91**
## General
**Before this PR**:
PRs have a template of about six questions. This is useful to prompt developers to think about testing, and the "Concerns" question can be used for flagging arbitrary concerns the users might have. However, it relies on developers identifying these concerns a priori.

**After this PR**:

==COMMIT_MSG==
PRs now have a longer template. This provides a number of prompts that are worth referencing, even if the writer may not have had thought about them at the time. Questions about testing and a general section for concerns remain.
==COMMIT_MSG==

**Priority**: This week, please.

**Concerns / possible downsides (what feedback would you like?)**:
- Is this too onerous? I think it shouldn't be, as questions that are irrelevant can be answered with N/A or just deleted. There may be cases in P0 situations where this is longer than desired, but there is no compulsion to fill this in in those cases. In general, I think this will serve as a forcing function to document changes in more detail (in the general case).

**Is documentation needed?**: 
No.

## Compatibility
**Does this PR create any API breaks (e.g. at the Java or HTTP layers) - if so, do we have compatibility?**:
No.

**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**:
No.

**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist?**:
No.

**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**:
No.

**Does this PR need a schema migration?**
No.

## Testing and Correctness
**What was existing testing like? What have you done to improve it?**:
N/A.

**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**:
N/A.

**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**:
N/A.

## Execution
**How would I tell this PR works in production? (Metrics, logs, etc.)**:
N/A... though we can measure how often this has been filled in on github.

**Has the safety of all log arguments been decided correctly?**:
N/A.

**Will this change significantly affect our spending on metrics or logs?**:
No.

**How would I tell that this PR does not work in production? (monitors, etc.)**:
People on the dev team should complain if this is causing problems.

**If this PR does not work as expected, how do I fix that state? Would rollback be straightforward?**:
Revert this commit; rollback is straightforward.

**If the above plan is more complex than “recall and rollback”, please tag the support PoC here**:
N/A.

## Scale
**Would this PR be expected to pose a risk at scale? Think of the shopping product at our largest stack.**:
No.

**Would this PR be expected to perform a large number of database calls, and/or expensive database calls (e.g., row range scans, concurrent CAS)?**:
No.

**Would this PR ever, with time and scale, become the wrong thing to do - and if so, how would we know that we need to do something differently?**:
... Maybe, though the onus is on the team to flag this. 
Also while I don't mind knowing about _every_ PR in this repository, I think @raiju probably won't want this for too long.

## Development Process
**Where should we start reviewing?**:
There is one file. Top down.

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**:
N/A.

**Please tag any other people who should be aware of this PR**:
(Don't have SV's OSS GH but I've sent this to her separately)
@raiju

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,17 +1,64 @@
...
+
+**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**:
+
+**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist?**:
+
+**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**:
+
+**Does this PR need a schema migration?**
+
+## Testing and Correctness
```
Comment (mdaudali) [inline .github/PULL_REQUEST_TEMPLATE.md:26]: Perhaps more broadly: What, if any, assumptions are made about the current SOTW? If they change over time, how will we find out? (In my mind, this is different to the scale question)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,17 +1,64 @@
...
 
+**Has the safety of all log arguments been decided correctly?**:
+
+**Will this change significantly affect our spending on metrics or logs?**:
+
+**How would I tell that this PR does not work in production? (monitors, etc.)**:
+
+**If this PR does not work as expected, how do I fix that state? Would rollback be straightforward?**:
+
+**If the above plan is more complex than “recall and rollback”, please tag the support PoC here**:
```
Comment (mdaudali) [inline .github/PULL_REQUEST_TEMPLATE.md:44]: (and for safety's sake, if it's the end of the week, tag the next PoC!)

**Review by mdaudali — APPROVED** [review]
lgtm, just a couple of thoughts / extensions that feel relevant

**Review by gsheasby — APPROVED** [review]
LGTM, let's start using this.

**Pull request merged**